src/huggingface_hub/file_download.py

```diff
@@ -1718,7 +1718,7 @@ def _download_to_tmp_and_move(
                 displayed_filename=filename,
             )
         else:
-            if xet_file_data is not None:
+            if xet_file_data is not None and not is_xet_disabled():
```

Comment:
small preference to not adding the helper `is_xet_disabled()` and just check the environment variable here and keep `is_xet_available()` unchanged
```suggestion
            if xet_file_data is not None and not constants._is_true(os.environ.get("HF_HUB_DISABLE_XET")):
```

Reply:
slight preference (on top of @hanouticelina 's comment) to have a constant named `HF_HUB_DISABLE_XET` in `constants.py` and check from there


```py
            if xet_file_data is not None and not constants.HF_HUB_DISABLE_XET:
```